jupyter_drives/managers/s3.py

```
@@ -0,0 +1,114 @@
+from s3contents import S3ContentsManager
+
+from ..base import DrivesConfig
+from .manager import JupyterDrivesManager
+
+
+class S3Manager(JupyterDrivesManager):
+    """Jupyter drives manager for S3 drives."""
+
+    def __init__(self, config: traitlets.config.Config) -> None:
+        super().__init__(DrivesConfig(config=config))
+        self._drives_cache = {}
+
+    @property
+    def base_api_url(self):
+        return self._config.api_base_url or "https://s3.amazonaws.com/"
+
+    @property
+    def per_page_argument(self) -> Optional[Tuple[str, int]]:
+        """Returns query argument to set number of items per page.
+
+        Returns
+            [str, int]: (query argument name, value)
+            None: the provider does not support pagination
+        """
+        return ("per_page", 100)
+    
+    async def list_drives(self) -> List[Dict[str, str]]:
+        """Get the list of available drives.
+            
+        Returns:
+            The list of available drives
+        """
+        # drives_url = "/get-listDrives"
+        # results = await self._call_s3(drives_url)  
+        
+        data = []
+        error = ""
+        if (self._config.access_key_id and self._config.secret_access_key):
+            S3Drive = get_driver(Provider.S3)
+            drives = [S3Drive(self._config.access_key_id, self._config.secret_access_key)]
+
+            results = []
+            for drive in drives:
+                results += drive.list_containers()
+        
+            for result in results:
+                data.append(
+                    {
+                        "name": result.name,
+                        "region": result.driver.region,
+                        "creation_date": result.extra["creation_date"],
+                        "status": "inactive",
+                        "provider": "S3"
+                    }
+                )
+        else:
+            error = "No AWS credentials provided."
```

We should raise an error instead

Similar to how we deal with missing credential in `_call_provider`